shell/edit/storage.k8s.io.storageclass/provisioners/bsu.csi.outscale.com.vue

```diff
@@ -0,0 +1,172 @@
+<script>
+import LabeledSelect from '@shell/components/form/LabeledSelect';
+import { SECRET_TYPES as TYPES } from '@shell/config/secret';
+import { LabeledInput } from '@components/Form/LabeledInput';
+import { RadioGroup } from '@components/Form/Radio';
+import UnitInput from '@shell/components/form/UnitInput';
+import { _CREATE } from '@shell/config/query-params';
+import { SECRET } from '@shell/config/types';
+
+export default {
+  components: {
+    LabeledSelect, LabeledInput, RadioGroup, UnitInput
+  },
+  props: {
+    value: {
+      type:     Object,
+      required: true
+    },
+    mode: {
+      type:     String,
+      required: true
+    }
+  },
+  async fetch() {
+    const secrets = await this.$store.dispatch('cluster/findAll', { type: SECRET });
+
+    this.secrets = secrets.filter(s => s._type === TYPES.OPAQUE).map(s => s.id).sort();
+  },
+  data() {
+    const canListStorageClasses = this.$store.getters['cluster/canList'](SECRET);
+    const volumeTypeOptions = [
+      {
+        label: this.t('storageClass.outscale-bsu.volumeType.gp2'),
+        value: 'gp2'
+      },
+      {
+        label: this.t('storageClass.outscale-bsu.volumeType.io1'),
+        value: 'io1'
+      },
+    ];
+
+    const availabilityZoneOptions = [
+      {
+        label: this.t('storageClass.outscale-bsu.availabilityZone.automatic'),
+        value: 'automatic'
+      },
+      {
+        label: this.t('storageClass.outscale-bsu.availabilityZone.manual'),
+        value: 'manual'
+      }
+    ];
+    const encryptionOptions = [
+      {
+        label: this.t('storageClass.outscale-bsu.encryption.enabled'),
+        value: 'true'
+      },
+      {
+        label: this.t('storageClass.outscale-bsu.encryption.disabled'),
+        value: 'false'
+      }
+    ];
+
+    if (this.mode === _CREATE) {
+      this.$set(this.value.parameters, 'type', this.value.parameters.type || volumeTypeOptions[0].value);
+      this.$set(this.value.parameters, 'encrypted', this.value.parameters.encrypted || encryptionOptions[1].value);
+      this.$set(this.value.parameters, 'iopsPerGB', this.value.parameters.iopsPerGB || '0');
+    }
+
+    return {
+      volumeTypeOptions,
+      availabilityZoneOptions,
+      encryptionOptions,
+      canListStorageClasses,
+      secrets: []
+    };
+  },
+  computed: {
+    iopsPerGB: {
+      get() {
+        return Number.parseInt(this.value.parameters.iopsPerGB);
+      },
+      set(value) {
+        this.$set(this.value.parameters, 'iopsPerGB', value.toString());
+      }
+    },
+    keySize: {
+      get() {
+        return Number.parseInt(this.value.parameters['luks-key-size']);
+      },
+      set(value) {
+        this.$set(this.value.parameters, 'luks-key-size', value);
+      }
+    },
+    keySecrets: {
+      set(value) {

```

Comment:
There's no `get` here, so when it's used in the `v-model` below there's lots of errors (on load and use)